src/openforms/submissions/form_logic.py

```diff
@@ -198,7 +200,7 @@ def check_submission_logic(
     if not submission_state.form_steps:
         return
 
-    rules = get_rules_to_evaluate(submission)
+    rules = get_rules_to_evaluate(submission, current_step)
```

Comment:
I was surprised (and suspicious) about this fix, is there a reason why previously the `current_step` was inferred for this flow, instead of explicitly passed to `get_rules_to_evaluate`?

Reply:
this function exists because the logic for the submission as a whole still needs to be evaluated during registration/processing after it's completed. In those contexts, there isn't a submission step, and it's also not relevant for the "trigger_from_step", because the assumption is that this only gets called for completed submissions that have gone through all the steps anyway.

Clearly this was a wrong assumption - as this does get called in the logic check serialization too which is in the context of a particular step. Nice find!